tests/test_client.py

```diff
@@ -263,6 +264,68 @@ def test_with_invalid_request():
         Client(should_validate_auth=False).submit(request)
 
 
+@responses.activate
+def test_get_request_has_user_agent_headers():
+    collection = Collection('foobar')
+    request = Request(
+        collection=collection,
+    )
+    responses.add(
+        responses.GET,
+        expected_submit_url(collection.id),
+        status=200,
+        json=expected_job(collection.id, 'abcd-1234'),
+    )
+
+    Client(should_validate_auth=False).submit(request)
+
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request is not None
+    assert responses.calls[0].request.headers is not None
+    headers = responses.calls[0].request.headers
+    assert "User-Agent" in headers
+    user_agent_header = headers["User-Agent"]
+    # Since it's kinda overkill to find the exact character set
+    #   allowed in platform/implementation/version/etc,
+    #   the following regex may be a little bit more tolerant
+    assert re.match(
+        r"\s*([^/\s]+/[^/\s]+)(\s+[^/\s]+/[^/\s]+)*\s*", user_agent_header
```

Comment:
Since you use this regex in a couple of tests it's probably best to define a constant so if we decide to change it in the future we just change it in one place.

Reply:
Good point ... will do.

Reply:
Generated user-agent header regex inside a function instead of using the same literals in multiple places.